app/src/androidTest/java/org/sagebionetworks/research/mpower/RoomScheduledActivityTests.kt

```diff
@@ -0,0 +1,239 @@
+        assertMedicationTaskReferenceActivity(roomActivityList?.first())
+    }
+
+    fun assertMedicationTaskReferenceActivity(activity: RoomScheduledActivity?) {
+        assertNotNull(activity)
+        assertEquals("273c4518-7cb6-4496-b1dd-c0b5bf291b09:2018-08-17T00:00:00.000", activity?.guid)
+        assertEquals("e24e6601-1822-48b0-8770-00870d870708", activity?.schedulePlanGuid)
+        assertEquals(false, activity?.persistent)
+        assertNotNull(activity?.scheduledOn)
+        assertEquals(DateTime.parse("2018-08-17T00:00-04:00").toDate().time, activity?.scheduledOn?.toDate()?.time)
+        assertNotNull(activity?.expiresOn)
+        assertEquals(DateTime.parse("2018-08-18T00:00-04:00").toDate().time, activity?.expiresOn?.toDate()?.time)
+        assertEquals(ScheduleStatus.EXPIRED, activity?.status)
+        assertNotNull(activity?.activity)
+        assertEquals("Medication", activity?.activity?.label)
+        assertEquals("273c4518-7cb6-4496-b1dd-c0b5bf291b09", activity?.activity?.guid)
+        assertNotNull(activity?.activity?.task)
+        assertEquals("Medication", activity?.activity?.task?.identifier)
+        assertNull(activity?.clientData?.data)
+    }
+
+    @Test
+    fun serialization_testSurveyReference() {
+        assertMotivationSurveyReferenceActivity(roomActivityList?.get(2))
+    }
+
+    fun assertMotivationSurveyReferenceActivity(activity: RoomScheduledActivity?) {
+        assertNotNull(activity)
+        assertEquals("a341c893-615d-48e1-ab6a-d418af720269:2018-08-17T13:40:39.183", activity?.guid)
+        assertNotNull(activity?.startedOn)
+        assertEquals(DateTime.parse("2018-08-17T17:44:05.114Z").toDate().time, activity?.startedOn?.toDate()?.time)
+        assertNotNull(activity?.finishedOn)
+        assertEquals(DateTime.parse("2018-08-17T18:09:28.610Z").toDate().time, activity?.finishedOn?.toDate()?.time)
+        assertNotNull(activity?.activity?.survey)
+        assertEquals("Motivation", activity?.activity?.survey?.identifier)
+        assertEquals("https://ws.sagebridge.org/v3/surveys/7beb8a71-65d0-4217" +
+                "-a4fb-3bb029b55258/revisions/2018-08-07T22:41:52.186Z", activity?.activity?.survey?.href)
+        assertEquals("{motivation_II_T=eat}", activity?.clientData?.data.toString())
+    }
+
+    @Test
+    fun query_testAll() {
+        reconfigureRoomDb()
+        val dbActivities = activityDao?.getAll()
+        assertNotNull(dbActivities)
+        assertEquals(6, dbActivities?.size)
+        assertMedicationTaskReferenceActivity(dbActivities?.first())
+        assertMotivationSurveyReferenceActivity(dbActivities?.get(2))
+    }
+
+    @Test
+    fun query_test() {
+        reconfigureRoomDb()
+        val dbActivities = activityDao?.getAll()
+        assertNotNull(dbActivities)
+        assertEquals(6, dbActivities?.size)
+        assertMedicationTaskReferenceActivity(dbActivities?.first())
+        assertMotivationSurveyReferenceActivity(dbActivities?.get(2))
+    }
```

Comment:
query_test() appears identical to query_testAll(). Is that a mistake?

Reply:
Yes it was, removed.